src/util.c

```diff
@@ -60,6 +61,23 @@ char *catsnprintf(char *strbuf, size_t *strbuf_len, const char *fmt, ...)
     return strbuf;
 }
 
+/* Checks return value of snprintf and panic on truncation or error. */
+int safesnprintf(void *buf, size_t size, const char *fmt, ...)
+{
+    va_list va;
+
+    va_start(va, fmt);
+    int ret = vsnprintf(buf, size, fmt, va);
+    va_end(va);
+
+    /* If size is zero, we assume this is intentional to calculate output len */
```

Comment:
Better to have a wrapper for length calculation, it will make the code more readable (and then you can remove this comment).

Reply:
Again, I don't like one liners much :) Once we learn snprintf() with zero buffer is used to calculate length, there won't be a readability problem. Probably, it'll be more readable than seeing a function, say `outputLength("%d", 100);` which you'll still need to go into its implementation to figure out what is happening.

Reply:
Well... of course we can remember that in order to get the length we can use this method with `0`, but I think it will be simple to have an explicit method for that purpose...

Reply:
Would `outputLen("%d", 100);"` make sense or `lensnprintf("%d, 100)` similar to `safesnprintf` and `catsnprintf` ? Any suggestion for the name? As it doesn't seem more readable to me, I cannot come up with a good name :)

Reply:
added lensnprintf()